core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2555,6 +2555,15 @@ private[spark] object Utils extends Logging {
       sparkJars.map(_.split(",")).map(_.filter(_.nonEmpty)).toSeq.flatten
     }
   }
+
+  private[util] val REDACTION_REPLACEMENT_TEXT = "*********(redacted)"
+  def redact(conf: SparkConf)(kv: (String, String)): (String, String) = {
+    val redactionPattern = conf.get(SECRET_REDACTION_PATTERN).r
```

Comment:
Compiling the regex once for every item in the list being redacted, instead of doing it once for the whole list.